services/sharedchannel/sync_recv.go

```diff
@@ -240,16 +247,23 @@ func (scs *Service) upsertSyncPost(post *model.Post, channel *model.Channel, rc
 			mlog.String("post_id", post.Id),
 			mlog.String("channel_id", post.ChannelId),
 		)
-	} else if rpost.DeleteAt == 0 {
+	} else if post.DeleteAt > 0 {
+		// delete post
+		rpost, appErr = scs.app.DeletePost(post.Id, post.UserId)
+		scs.server.GetLogger().Log(mlog.LvlSharedChannelServiceDebug, "Deleted sync post",
+			mlog.String("post_id", post.Id),
+			mlog.String("channel_id", post.ChannelId),
+		)
+	} else if post.EditAt > rpost.EditAt || post.Message != rpost.Message {
```

Comment:
Interesting. Are we being extra cautious or is it actually possible for the content to have changed and not `EditAt`?

Reply:
Just being cautious.  I hit so many corner cases debugging this stuff that I decided not to trust any contracts.